### ISSUE TITLE
Detect when managed toolchains are in use during uninstall

### ISSUE THREAD

**Issue by zanieb** [astral-sh/uv#4718]
We should track our uses of managed toolchains (e.g. when used in `uv venv`, `uv tool install` and `uv sync`) then ensure that the toolchain is no longer in use before removing it during uninstall or require opt-in to force removal.

**Comment by aretrace** [comment]
Could tools invoke a separate interpreter copy (e.g., `cpython-3.12.8-macos-aarch64-none-<TOOL_NAME>`)?
An additional option for interpreter selection during tool installation would make sense  under this scenario.

**Comment by zanieb** [comment]
Yeah we could have each tool get a dedicated interpreter, but I don't think that's ideal unless we can use hard-links?

I would rather:

- Not allow uninstall unless you change the tools (or use a force flag)
- Make it easy to change tools to another Python version
- Link tools to a "stable" directory of minor Python interpreter versions so patch upgrades do not affect them

**Comment by aretrace** [comment]
👆 yes
